### DISCUSSION TITLE
update Services schema to allow lossless import/export

### DISCUSSION THREAD

**Discussion by michielbdejong** [tosdr/edit.tosdr.org#1048] [Ideas]
The current Services schema cannot hold all information held by the [Services API spec](https://github.com/tosdr/tosdr.org/wiki/Specification:-services):

string fields to add:
* id -> add `oldId`
* keywords, related -> add

already have:
* name -> lowercase to `name`
* urls -> use `url` (there are actually no services in the db with multiple urls, that's already split)
* tosdr.rated -> use `grade`, already a string

don't need:
* type -> restrict to 'service'
* fulltos -> deprecate (link to tosback!)
* parent, license -> deprecate
* dataexport, freesoftware -> make point if yes
* meta.spec-version -> always 1.1
* alexa -> check on the fly

**Comment by michielbdejong** [comment]
Calling oldId 'slug' to avoid camelCase.

**Comment by michielbdejong** [comment]
Should make ../tosdr-build/src/services same format as old_db/services/

**Comment by evilcel3ri** [comment]
> keywords, related -> add

Is that needed ? What kind of keywords would be used ? "Chat", "email", etc ?

**Comment by michielbdejong** [comment]
Yes, otherwise e.g. https://tosdr.org/#search=shop would stop working
